internal/core/cloud/metadata_test.go

```diff
@@ -302,6 +311,31 @@ func (suite *CloudMetadataTestSuite) TestNewCloudInstanceAWS() {
 	suite.Equal("some-id", meta.InstanceID)
 }
 
+func (suite *CloudMetadataTestSuite) TestNewAWSCloudInstanceWithoutMetadata() {
+	ctx := context.TODO()
+	suite.mockExecutor.
+		On("Exec", "dmidecode", "-s", "chassis-asset-tag").
+		Return(dmidecodeEmpty(), nil).
+		On("Exec", "dmidecode", "-s", "system-version").
+		Return(dmidecodeAWSSystem(), nil)
```

Comment:
In order to improve decoupling and testability provider detection needs to be refactored, imho, by possibly introducing an interface.

However this would be out of scope here and provider detection is going to be evolved anyways when we address PPC support.
[A gist on what I mean here](https://github.com/trento-project/agent/pull/413/files#diff-f858dae8a26664ac95b63cfea1249fbcb07edec4be48e4d7f064d2f839d5a4a1R20)